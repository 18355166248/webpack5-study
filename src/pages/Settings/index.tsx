import { FC } from "react";
import * as styles from "./index.module.scss";
import { Button, Input, message } from "antd";
import { useNavigate } from "react-router-dom";
import cn from "classnames";
import { isFunction } from "../../utils/base";

interface SettingsProps {}

const Settings: FC<SettingsProps> = () => {
  const navigate = useNavigate();

  function back() {
    navigate(-1);
  }
  function setting() {
    isFunction(message.success) && message.success("设置成功");
  }
  return (
    <div className={cn(styles.settings, "mt-10 pl-10")}>
      <div className="mb-10">
        <Input placeholder="a query by placeholder t" />
      </div>
      <div className="mb-10">
        <Button onClick={back}>返回</Button>
      </div>
      <div>
        <Button type="primary" onClick={setting}>
          设置
        </Button>
      </div>
    </div>
  );
};

export default Settings;
