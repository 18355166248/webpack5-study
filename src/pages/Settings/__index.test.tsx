import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Settings from "./index";
import { message } from "antd";
import userEvent from "@testing-library/user-event";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => jest.fn(),
}));

jest.mock("antd", () => ({
  ...jest.requireActual("antd"),
  message: {
    success: jest.fn(),
  },
}));

describe("Settings test", () => {
  test("Settings render snap", () => {
    const { baseElement } = render(<Settings />);
    expect(baseElement).toMatchSnapshot();
  });
  test("first Settings test", () => {
    render(<Settings />);

    const placeholderInput = screen.getByPlaceholderText(
      "a query by placeholder t"
    );
    expect(placeholderInput).toBeInTheDocument();
  });

  test("Settings click back", () => {
    const mockNavigate = jest.fn();
    jest
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      .spyOn(require("react-router-dom"), "useNavigate")
      .mockReturnValue(mockNavigate);

    render(<Settings />);

    const backButton = screen.getByRole("button", { name: /返 回/i });
    fireEvent.click(backButton);

    expect(mockNavigate).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });

  test("Settings click 设置", async () => {
    const user = userEvent.setup();
    render(<Settings />);

    const backButton = screen.getByRole("button", { name: /设 置/i });
    // 这里需要await 不然后面的断言无法触发
    await user.click(backButton);

    expect(message.success).toHaveBeenCalled();
    expect(message.success).toHaveBeenCalledTimes(1);
    expect(message.success).toHaveBeenCalledWith("设置成功");
  });
});
