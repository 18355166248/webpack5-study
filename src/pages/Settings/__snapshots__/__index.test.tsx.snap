// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings test Settings render snap 1`] = `
<body>
  <div>
    <div
      class="mt-10 pl-10"
    >
      <div
        class="mb-10"
      >
        <input
          class="ant-input css-dev-only-do-not-override-d2lrxs ant-input-outlined"
          placeholder="a query by placeholder t"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-10"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-d2lrxs ant-btn-default"
          type="button"
        >
          <span>
            返 回
          </span>
        </button>
      </div>
      <div>
        <button
          class="ant-btn css-dev-only-do-not-override-d2lrxs ant-btn-primary"
          type="button"
        >
          <span>
            设 置
          </span>
        </button>
      </div>
    </div>
  </div>
</body>
`;
